Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -1781,6 +1782,18 @@ void CGame::Packet_PlayerJoinData(CPlayerJoinDataPacket& Packet)
         }
 #endif
 
+        // Prevent player from connecting if serial is invalid
+        std::regex serialRegex("^[A-F0-9]{32}$");
+        if (!std::regex_match(strSerial, serialRegex))
+        {
+            // Tell the console
+            CLogger::LogPrintf("CONNECT: %s failed to connect (Invalid serial) (%s)\n", szNick, pPlayer->GetSourceIP());
```

Comment:
Does `GetSourceIP()` return `std::string`/`SString` or `const char*`? If the former then you need to use `.c_str()`

Reply:
sorry that was incorrect, check now